packages/teams-js/test/public/settings.spec.ts

```diff
@@ -71,46 +57,131 @@ describe('settings', () => {
     });
   });
 
-  it('should successfully get settings', done => {
-    utils.initializeWithContext('settings').then(() => {
-      settings.getSettings(settings => {
-        expect(settings).toBe(expectedSettings);
-        done();
-      });
-
-      const message = utils.findMessageByFunc('settings.getSettings');
-      expect(message).not.toBeNull();
-
+  describe('Testing settings.getSettings function', () => {
+    const allowedContexts = [
+      FrameContexts.content,
+      FrameContexts.settings,
+      FrameContexts.remove,
+      FrameContexts.sidePanel,
+    ];
+    const expectedSettings: settings.Settings = {
+      suggestedDisplayName: 'someSuggestedDisplayName',
+      contentUrl: 'someContentUrl',
+      websiteUrl: 'someWebsiteUrl',
+      entityId: 'someEntityId',
+    };
+
+    it('settings.getSettings should not allow calls before initialization', () => {
       const expectedSettings: settings.Settings = {
         suggestedDisplayName: 'someSuggestedDisplayName',
         contentUrl: 'someContentUrl',
         websiteUrl: 'someWebsiteUrl',
         entityId: 'someEntityId',
       };
+      expect(() => {
+        settings.getSettings(settings => {
+          expect(settings).toBe(expectedSettings);
+        });
+      }).toThrowError('The library has not yet been initialized');
+    });
 
-      utils.respondToMessage(message, expectedSettings);
+    Object.values(FrameContexts).forEach(context => {
+      if (!allowedContexts.some(allowedContexts => allowedContexts == context)) {
+        it(`settings.getSettings does not allow calls from ${context} context`, async () => {
+          await utils.initializeWithContext(context);
+          expect(() => {
+            settings.getSettings(settings => {
+              expect(settings).toBe(expectedSettings);
```

Comment:
I don't think this is correct. We don't want to validate that the settings are correct in the tests where we expect it to throw an error. I'm concerned that the tests are passing with both "expect success" and "expect error" in the same block of code. If we get to line 94, the test should fail.

Reply:
In fact, if you could add an expect that fails if this line of code is reached, that would be preferable.

Reply:
I fix it by sending empty callback function can you please check this in the latest commit.